packages/@lwc/engine-dom/src/styles.ts

```diff
@@ -65,42 +76,37 @@ function createFreshStyleElement(content: string) {
     return elm;
 }
 
-function createStyleElement(content: string) {
-    // For a mysterious reason, IE11 doesn't like the way we clone <style> nodes
-    // and will render the incorrect styles if we do things that way. It's just
-    // a perf optimization, so we can skip it for IE11.
-    if (isIE11) {
-        return createFreshStyleElement(content);
-    }
-
-    let elm = styleElements.get(content);
-    if (isUndefined(elm)) {
-        // We don't clone every time, because that would be a perf tax on the first time
-        elm = createFreshStyleElement(content);
-        styleElements.set(content, elm);
-    } else {
+function createStyleElement(content: string, cacheData: StyleElementCacheData) {
+    const { stylesheet, used } = cacheData;
+    if (used) {
+        // For a mysterious reason, IE11 doesn't like the way we clone <style> nodes
+        // and will render the incorrect styles if we do things that way. It's just
+        // a perf optimization, so we can skip it for IE11.
+        if (isIE11) {
+            return createFreshStyleElement(content);
+        }
         // This `<style>` may be repeated multiple times in the DOM, so cache it. It's a bit
         // faster to call `cloneNode()` on an existing node than to recreate it every time.
-        elm = elm.cloneNode(true) as HTMLStyleElement;
+        return stylesheet.cloneNode(true) as HTMLStyleElement;
     }
-    return elm;
+    // We don't clone every time, because that would be a perf tax on the first time
+    cacheData.used = true;
+    return stylesheet;
 }
 
-function createOrGetConstructableStylesheet(content: string) {
-    // It's important for CSSStyleSheets to be unique based on their content, so
-    // that adoptedStyleSheets.indexOf(sheet) works
-    let stylesheet = stylesheets.get(content);
-    if (isUndefined(stylesheet)) {
-        stylesheet = new CSSStyleSheet();
-        stylesheet.replaceSync(content);
-        stylesheets.set(content, stylesheet);
-    }
+function createConstructableStylesheet(content: string) {
+    const stylesheet = new CSSStyleSheet();
+    stylesheet.replaceSync(content);
     return stylesheet;
 }
 
-function insertConstructableStylesheet(content: string, target: ShadowRoot | Document) {
-    const stylesheet = createOrGetConstructableStylesheet(content);
+function insertConstructableStylesheet(
+    content: string,
+    target: ShadowRoot | Document,
```

Comment:
There is a specific interface for this.

```suggestion
    target: DocumentOrShadowRoot,
```

Reply:
Interestingly, this doesn't work. TypeScript complains that `Property 'appendChild' does not exist on type 'DocumentOrShadowRoot'`. Somehow `Document | ShadowRoot` is different from `DocumentOrShadowRoot`, at least in TypeScript. 😛